noise.zig

```diff
@@ -0,0 +1,668 @@
+const std = @import("std");
+const crypto = std.crypto;
+const mem = std.mem;
+const testing = std.testing;
+const expect = testing.expect;
+
+const Ed25519 = crypto.sign.Ed25519;
+const X25519 = crypto.dh.X25519;
+const Blake2s256 = crypto.hash.blake2.Blake2s256;
+const ChaCha20Poly1305 = crypto.aead.chacha_poly.ChaCha20Poly1305;
+
+pub const HASH_LEN = 32;
```

Comment:
A small nit, though constants should be defined in lower-cased snake case. https://ziglang.org/documentation/master/#Names